common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -238,8 +243,78 @@ def student_view(self, context):
+        """
+        Capture basic information about this sequence in New Relic before
+        rendering it. We do this up here because proctored exam functionality
+        could short circuit rendering after this section.
+        """
+        newrelic.agent.add_custom_parameter('seq.block_id', unicode(self.location))
+        newrelic.agent.add_custom_parameter('seq.display_name', self.display_name or '')
+        newrelic.agent.add_custom_parameter('seq.position', self.position)
+        newrelic.agent.add_custom_parameter('seq.is_time_limited', self.is_time_limited)
+
+    def _capture_full_seq_item_metrics(self, display_items):
+        """
+        Capture information about the number and types of XBlock content in
+        the sequence as a whole. We send this information to New Relic so that
+        we can do better performance analysis of courseware.
+        """
+        # Basic count of the number of Units (a.k.a. VerticalBlocks) we have in
+        # this learning sequence
+        newrelic.agent.add_custom_parameter('seq.num_units', len(display_items))
+
+        # Count of all modules (leaf nodes) in this sequence (e.g. videos,
+        # problems, etc.) The units (verticals) themselves are not counted.
+        all_item_keys = self._locations_in_subtree(self)
+        newrelic.agent.add_custom_parameter('seq.num_items', len(all_item_keys))
+
+        # Count of all modules by block_type (e.g. "video": 2, "discussion": 4)
+        block_counts = collections.Counter(usage_key.block_type for usage_key in all_item_keys)
+        for block_type, count in block_counts.items():
+            newrelic.agent.add_custom_parameter('seq.block_counts.{}'.format(block_type), count)
+
+    def _capture_current_unit_metrics(self, display_items):
+        """
+        Capture information about the current selected Unit within the Sequence.
+        """
+        # Positions are stored with indexing starting at 1. If we get into a
+        # weird state where the saved position is out of bounds (e.g. the
+        # content was changed), avoid going into any details about this unit.
+        if 1 <= self.position <= len(display_items):
+            # Basic info about the Unit...
+            current = display_items[self.position - 1]
+            newrelic.agent.add_custom_parameter('seq.current.block_id', unicode(current.location))
+            newrelic.agent.add_custom_parameter('seq.current.display_name', current.display_name or '')
+
+            # Examining all items inside the Unit (or split_test, conditional, etc.)
+            child_locs = self._locations_in_subtree(current)
+            newrelic.agent.add_custom_parameter('seq.current.num_items', len(child_locs))
+            curr_block_counts = collections.Counter(usage_key.block_type for usage_key in child_locs)
+            for block_type, count in curr_block_counts.items():
+                newrelic.agent.add_custom_parameter('seq.current.block_counts.{}'.format(block_type), count)
```

Comment:
You can see an example in the transaction trace screenshot near the top of this PR.